snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/SnomedRelationshipSearchRequest.java

```diff
@@ -63,6 +71,18 @@
 		return SnomedRelationshipIndexEntry.class;
 	}
 	
+	@Override
+	protected void collectAdditionalFieldsToFetch(Builder<String> additionalFieldsToLoad) {
+		if (fields().contains(SnomedRelationship.Fields.VALUE)) {
+			additionalFieldsToLoad.add(SnomedRelationshipIndexEntry.Fields.VALUE_TYPE);
+		}
+	}
+	
+	@Override
+	protected Multimap<String, String> collectFieldsToLoadReplacements() {
+		return REPLACE_VALUE_FIELD;
+	}
+	
```

Comment:
How about adding `SnomedRelationshipIndexEntry.Fields.VALUE_TYPE` as a fourth value for key `SnomedRelationship.Fields.VALUE` in the Multimap? That should handle both cases.

Reply:
Hmm, true 😃